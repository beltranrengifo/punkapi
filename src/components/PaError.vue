<template>
  <section class="pa-error text-center">
    <h2 class="mb-8">
      {{ error }}
    </h2>
    <p>
      Not funny :(
    </p>
  </section>
</template>

<script>
export default {
  name: 'PaError',

  props: {
    error: {
      type: Error,
      required: true,
    },
  },
}
</script>
