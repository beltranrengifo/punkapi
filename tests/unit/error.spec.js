
import { shallowMount } from '@vue/test-utils'
import PaError from '@/components/PaError'

describe('Error component', () => {
  let wrapper
  const error = new Error('Not funny')

  beforeEach(() => {
    wrapper = shallowMount(PaError, {
      propsData: {
        error,
      },
    })
  })

  it('Component should find the title', () => {
    expect(wrapper.find('h2')).toBeTruthy()
  })

  it('Component should render the error', () => {
    expect(wrapper.find('h2').text(error.toString())).toBeTruthy()
  })

  it('Component should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
