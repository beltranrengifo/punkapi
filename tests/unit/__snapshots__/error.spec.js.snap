// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component Component should match the snapshot 1`] = `
<section
  class="pa-error text-center"
>
  <h2
    class="mb-8"
  >
    
    Error: Not funny
  
  </h2>
   
  <p>
    
    Not funny :(
  
  </p>
</section>
`;
